### PR TITLE
Update OTA queues to use the matter work queue.

### LINKED ISSUE #20591: [OTA Matter.framework] Code is unsafe/racy error message
#### Problem
After setting the OTA delegate in Matter.framework and receiving Query Image InvokeCommand to the controller (Matter.framework), an error message states that the code is unsafe/racy. This appears to be the issue because during the transmission of the message we are using a delegate that is passed in from the caller [here](https://github.com/project-chip/connectedhomeip/pull/20516/files#diff-13bd030c2b38e3e7d4738070dd3c919bbdfe9ec8b39cc29996989936150e6d1fR59).  After inspecting the code, the assertion calls IsChipStackLockedByCurrentThread to check whether messaging is locked. For this to be set to true the following condition needs to be true `!mWorkQueue || mIsWorkQueueSuspended || dispatch_get_current_queue() == mWorkQueue` 

 #### Proposed Solution
Update the queue used to send the message to the node. 

[controller_error_log.txt](https://github.com/project-chip/connectedhomeip/files/9089424/controller_error_log.txt)

### PR DESCRIPTION
**Pull request by krypton36**
#### Problem
There is a racy/unsafe scenario when using OTA software update in the Matter.framework. 
- Fixes #20591 

#### Change overview
- Updates the dispatch queue to use the matter work queue.

#### Testing
- Ran scenario that causes this issue and no longer reproduced.

### CODE REVIEW

File: src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm [hunk 1]
```diff
@@ -28,15 +29,10 @@
...
 {
-    if (delegate && queue) {
-        mDelegate = delegate;
-        mQueue = queue;
-    } else {
-        mDelegate = nil;
-        mQueue = nil;
-    }
+    mDelegate = delegate ?: nil;
+    mQueue = chip::DeviceLayer::PlatformMgrImpl().GetWorkQueue();
```
Comment (bzbarsky-apple) [inline src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm:35]: This means the delegate will get called back on the Matter queue, right?

At the very least this needs to be clearly documented in the API documentation (by saying that the delegate will get called back on some random queue and might need to proxy things over to the right queue itself).

File: src/darwin/Framework/CHIP/MTRDeviceController.h [hunk 2]
```diff
@@ -124,10 +124,8 @@ typedef void (^MTRDeviceConnectionCallback)(MTRBaseDevice * _Nullable device, NS
  * Set the Delegate for the OTA Provider as well as the Queue on which the Delegate callbacks will be triggered
```
[the remark this answers is not in the document]
Comment (krypton36) [reply src/darwin/Framework/CHIP/MTRDeviceController.h:124]: This is no longer setting the queue. I will update the description. thank you.

**Review by carol-apple — APPROVED** [review]

**Review by bzbarsky-apple — APPROVED** [review]

File: src/darwin/Framework/CHIP/MTRDeviceController.h (hunk 2 above)
[the remark this answers is not in the document]
Comment (krypton36) [reply src/darwin/Framework/CHIP/MTRDeviceController.h]: We are back to allowing the caller set the queue.

File: src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm (hunk 1 above)
Comment (krypton36) [reply src/darwin/Framework/CHIP/MTROTAProviderDelegateBridge.mm]: We are back to allowing the caller set the queue.

**Review by andy31415 — APPROVED** [review]

**Pull request merged**